### PR TITLE
[SofaConstraint]: add a force data field in SlidingConstraint

### PR DESCRIPTION
**Pull request by TMorzadec**
This data holds the force applied to satisfy the SlidingConstraint

Rename some data with more explicit names following Sofa conventions

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by TMorzadec** [comment]
You are welcome! 

Are you interested in SOFA?

File: modules/SofaConstraint/SlidingConstraint.h [hunk 1]
```diff
@@ -66,6 +67,7 @@ class SlidingConstraint : public core::behavior::PairInteractionConstraint<DataT
     Data<int> m1; ///< index of the spliding point on the first model
     Data<int> m2a; ///< index of one end of the sliding axis
     Data<int> m2b; ///< index of the other end of the sliding axis
+    Data<Deriv> mforce; ///< interaction force
```
Comment (guparan) [inline modules/SofaConstraint/SlidingConstraint.h]: Please use `d_` prefix for Data members even if it's not (yet) consistent with existing Data members.
See https://github.com/sofa-framework/sofa/blob/master/GUIDELINES.md#naming

File: modules/SofaConstraint/SlidingConstraint.h [hunk 2]
```diff
@@ -46,6 +46,7 @@ class SlidingConstraint : public core::behavior::PairInteractionConstraint<DataT
     typedef typename DataTypes::VecCoord VecCoord;
     typedef typename DataTypes::VecDeriv VecDeriv;
     typedef typename DataTypes::MatrixDeriv MatrixDeriv;
+    //typedef typename DataTypes::MultiVecDerivId MultiVecDerivId;
```
Comment (guparan) [inline modules/SofaConstraint/SlidingConstraint.h]: To be removed if not needed.

**Review by guparan — CHANGES_REQUESTED** [review]

File: modules/SofaConstraint/SlidingConstraint.inl [hunk 3]
```diff
@@ -63,39 +63,40 @@ void SlidingConstraint<DataTypes>::buildConstraintMatrix(const core::ConstraintP
     const Coord B = x2.getValue()[tm2b];
 
     // the axis
-    Coord uniAB = B - A;
-    const Real ab = uniAB.norm();
-    uniAB.normalize();
+
+    Deriv m_dirAxe = B - A;
```
Comment (damienmarchal) [inline modules/SofaConstraint/SlidingConstraint.inl]: This line redefine m_dirAxe which is already defined as a class member.
So the change on this function on m_dirAxe are not stored into the class member but only on this local variable. 

The consequence is that in the line in the store lambda function:
```cpp
 mforce.setValue( m_dirProj* lamb1 + m_dirOrtho * lamb2 + m_dirAxe * lamb3);
```
is not working as expected as the m_dirAxe never change.

**Comment by damienmarchal** [comment]
@guparan, @hugtalbot  the bug was fixed. Do we set it to ready without waiting next week ?

**Comment by guparan** [comment]
Sure, feel free to do it if you reviewed the latest changes.
Let's not forget the [ci-build][with-scene-tests] though :wink:

**Comment by damienmarchal** [comment]
Hey...this PR has not been reviewed since the last request for change. Anyone can do that.

**Comment by guparan** [comment]
[ci-build][with-scene-tests]

**Pull request merged**